### PR TITLE
Update CCEGLView.cpp

### PR DESCRIPTION
**Pull request by FlagellumDei**
centerWindow() will not center the window correctly when the taskbar is located along the top or right screen edge.

### CODE REVIEW

File: cocos2dx/sprite_nodes/CCSpriteFrameCache.cpp [hunk 1]
```diff
@@ -242,6 +242,20 @@ void CCSpriteFrameCache::addSpriteFramesWithFile(const char *pszPlist)
         {
             // try to read  texture file name from meta data
             texturePath = metadataDict->valueForKey("textureFileName")->getCString();
+            
+            // Zwoptex hack: if we want to use a PVR texture it won't load because texturePath
+            // is empty, and cocos2dx will try to load the texture as PNG (which doesn't exists)
+            // It seems Zwoptex stores the texture file name and extension in "target".
+            if (texturePath.empty())
+            {
+                CCDictionary* targetDict = (CCDictionary*)metadataDict->objectForKey("target");
```
Comment (dumganhar) [inline cocos2dx/sprite_nodes/CCSpriteFrameCache.cpp:251]: This change didn't match the title of this Pull Request. Please drop it from this PR, and send it in another PR, thanks.

Comment (FlagellumDei) [inline cocos2dx/sprite_nodes/CCSpriteFrameCache.cpp:251]: Sorry, this is my first pull request. How am I supposed to drop CCSpriteFrameCache update from the request without closing the whole request?

Comment (dumganhar) [inline cocos2dx/sprite_nodes/CCSpriteFrameCache.cpp:251]: Hi, the way of separating it to 2 pull requests is : 

if your current branch is patch-1
- Step 1: 

`git checkout -b clean-branch` 
`git reset --hard 1e534dd1dd442817112e0376c8075e7b405378af` // reset commits before your changes, this will make branch `clean-branch` to be real clean branch.
- Step 2: Create a new branch for changes of `centerWindow`

`git checkout -b  patch-center` // create a new branch based on clean branch
`git cherry-pick 503d49d826863d3c46872045502d259dfa5214d4` // pick commit of "Update CCEGLView.cpp"
`git push origin patch-center:patch-1` // push to your origin remote repo, this pull request will be automatically updated.
- Step 3: Send a new pull request for changes of CCSpriteFrameCache.

`git checkout clean-branch`
`git checkout -b patch-2`
`git cherry-pick 155e1df30c40c62bc5343e5a2d0dcf5eedd29f82` // pick commit of "Update CCSpriteFrameCache.cpp"
`git push origin patch-2` // Then you only need to send this Pull Request for this branch.
